### PR TITLE
paging, item, and name context fixes

### PR DESCRIPTION
**Pull request by ethn**

### CODE REVIEW

File: card/lib/card/env.rb [hunk 1]
```diff
@@ -45,19 +45,12 @@ def slot_opts
         # FIXME:  upgrade to safe parameters
         self[:slot_opts] ||= begin
           opts = params[:slot] || {}
-          opts.merge shortcut_slot_opts
+          opts.merge! shortcut_slot_opts
```
Comment (ethn) [inline card/lib/card/env.rb]: important bang.

Comment (ethn) [reply card/lib/card/env.rb]: with this fixed, we can get rid of all the other references to `params[:item]`

File: card/mod/standard/set/all/rich_html/show.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ def show_method
   def show_without_page_layout view, args
     @main = true if params[:is_main] || args[:main]
     args.delete(:layout)
-    view ||= args[:home_view] || :open
+    view ||= args[:home_view] || default_nest_view
```
Comment (ethn) [inline card/mod/standard/set/all/rich_html/show.rb]: this is debatable.

File: card/mod/standard/set/all/rich_html/process_layout.rb [hunk 3]
```diff
@@ -28,6 +28,7 @@ def page_layout
   def render_with_layout view, layout, args={}
     view_opts = Layout.main_nest_opts(layout, self)
     view ||= view_opts.delete(:view) || default_nest_view
+    view_opts[:home_view] = view
```
Comment (ethn) [inline card/mod/standard/set/all/rich_html/process_layout.rb]: needed for paging.

File: card/lib/card/env.rb (hunk 1 above)
Comment (xithan) [reply card/lib/card/env.rb]: If you want to manipulate params maybe you should do it less concealed.

Comment (ethn) [reply card/lib/card/env.rb]: I don't think we need to manipulate params.  I'll change this to clone.

**Pull request merged**